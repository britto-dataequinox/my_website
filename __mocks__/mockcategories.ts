export const mockcategories = [
  {
    id: 'cat_9BAmwJmb3leXdn',
    parent_id: null,
    slug: 'electronics',
    name: 'Electronics',
    description: '1. Laptops\n2.Mobiles\n3.Ipods, Ipads\n4.Headsets',
    products: 3,
    created: 1708105746,
    updated: 1708105746,
    meta: null,
    assets: [
      {
        id: 'ast_QG375vdYKYwrMO',
        url: 'https://cdn.chec.io/merchants/56103/assets/ZjaBIGXVDG4kh94r|61uNK7su24L._SL1500_.jpg',
        description: null,
        is_image: true,
        filename: '61uNK7su24L._SL1500_.jpg',
        file_size: 18709,
        file_extension: 'jpg',
        image_dimensions: {
          width: 1500,
          height: 1500,
        },
        meta: [],
        created_at: 1708105701,
        updated_at: 1708105703,
      },
      {
        id: 'ast_RqEv5xYDG7lZz4',
        url: 'https://cdn.chec.io/merchants/56103/assets/zGUfgJDMGb8TFdnI|71jG+e7roXL._SL1500_.jpg',
        description: null,
        is_image: true,
        filename: '71jG+e7roXL._SL1500_.jpg',
        file_size: 40326,
        file_extension: 'jpg',
        image_dimensions: {
          width: 1500,
          height: 1500,
        },
        meta: [],
        created_at: 1708105735,
        updated_at: 1708105737,
      },
    ],
    children: [],
  },
];
