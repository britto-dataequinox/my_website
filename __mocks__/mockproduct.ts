export const mockproducts = [
  {
    id: 'prod_kpnNwADxLzlmXB',
    created: 1707974547,
    updated: 1708163913,
    active: true,
    permalink: 'UxeBoj',
    name: 'Apple MacBook Air',
    description:
      '<p></p><ul><li><p>All-Day Battery Life – Go longer than ever with up to 18 hours of battery life.</p></li><li><p>Powerful Performance – Take on everything from professional-quality editing to action-packed gaming with ease. The Apple M1 chip with an 8-core CPU delivers up to 3.5x faster performance than the previous generation while using way less power.</p></li><li><p>Superfast Memory – 8GB of unified memory makes your entire system speedy and responsive. That way it can support tasks like memory-hogging multitab browsing and opening a huge graphic file quickly and easily.</p></li><li><p>Stunning Display – With a 13.3-inch/33.74 cm Retina display, images come alive with new levels of realism. Text is sharp and clear, and colors are more vibrant.</p></li><li><p>Why Mac – Easy to learn. Easy to set up. Astoundingly powerful. Intuitive. Packed with apps to use right out of the box. Mac is designed to let you work, play, and create like never before.</p></li></ul>',
    price: {
      raw: 1011,
      formatted: '1,011.00',
      formatted_with_symbol: '$1,011.00',
      formatted_with_code: '1,011.00 USD',
    },
    inventory: {
      managed: true,
      available: 50,
    },
    sku: null,
    sort_order: 0,
    seo: {
      title: null,
      description: null,
    },
    thank_you_url: null,
    meta: null,
    conditionals: {
      is_active: true,
      is_tax_exempt: false,
      is_pay_what_you_want: false,
      is_inventory_managed: true,
      is_sold_out: false,
      has_digital_delivery: false,
      has_physical_delivery: true,
      has_images: true,
      collects_fullname: false,
      collects_shipping_address: true,
      collects_billing_address: false,
      collects_extra_fields: false,
    },
    is: {
      active: true,
      tax_exempt: false,
      pay_what_you_want: false,
      inventory_managed: true,
      sold_out: false,
    },
    has: {
      digital_delivery: false,
      physical_delivery: true,
      images: true,
    },
    collects: {
      fullname: false,
      shipping_address: true,
      billing_address: false,
      extra_fields: false,
    },
    checkout_url: {
      checkout: 'https://checkout.chec.io/UxeBoj?checkout=true',
      display: 'https://checkout.chec.io/UxeBoj',
    },
    categories: [],
    image: {
      id: 'ast_gvRjwObO3ml4mN',
      url: 'https://cdn.chec.io/merchants/56103/assets/6uEp0drDg6pAUn5h|macbook.jpg',
      description: null,
      is_image: true,
      filename: 'macbook.jpg',
      file_size: 40326,
      file_extension: 'jpg',
      image_dimensions: {
        width: 1500,
        height: 1500,
      },
      meta: [],
      created_at: 1708147007,
      updated_at: 1708147010,
    },
  },
];
