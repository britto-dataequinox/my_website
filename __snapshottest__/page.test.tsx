import Home from '@/pages';
import React from 'react';
import renderer from 'react-test-renderer';
import { mockproducts } from '../__mocks__/mockproduct';
import { mockcategories } from '../__mocks__/mockcategories';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/',
      pathName: '',
      query: '',
      asPath: '/',
      push: jest.fn(),
    };
  },
}));

test('page snap', () => {
  const tree = renderer
    .create(<Home products={mockproducts} categories={mockcategories} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
