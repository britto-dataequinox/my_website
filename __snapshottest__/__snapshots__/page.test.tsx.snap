// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page snap 1`] = `
<div
  className="container"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1bb29ia-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 css-1bn9r4q-MuiTypography-root"
        onClick={[Function]}
      >
        Online Store
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          href="/cart"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <span
            className="MuiBadge-root css-1c32n2y-MuiBadge-root"
            data-testid="cart-badge"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingCartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
            <span
              className="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-kf66pb-MuiBadge-badge"
            />
          </span>
        </a>
      </button>
    </div>
  </header>
  <div
    className="MuiBox-root css-y1gt6f"
  >
    <h2
      style={
        {
          "textAlign": "center",
        }
      }
    >
      All Products
    </h2>
  </div>
  <main
    className="main"
  >
    <div
      className="MuiBox-root css-xrjolu"
    >
      <div
        className="MuiInputBase-root [object Object] MuiInputBase-colorPrimary css-1le28he-MuiInputBase-root"
        onClick={[Function]}
      >
        <input
          aria-label="search"
          className="MuiInputBase-input [object Object] css-yz9k0d-MuiInputBase-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="search..."
          type="text"
          value=""
        />
      </div>
      <button
        aria-label="search"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
          />
        </svg>
      </button>
    </div>
  </main>
  <div
    className="MuiBox-root css-1esmyhx"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10u79dg-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
          >
            Categories
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <ul
          style={
            {
              "listStyle": "none",
              "padding": 0,
            }
          }
        >
          <li>
            <label
              className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
            >
              <span
                className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                onBlur={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <input
                  checked={false}
                  className="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate={false}
                  onChange={[Function]}
                  required={false}
                  type="checkbox"
                />
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
              >
                Electronics
              </span>
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 css-14j1sok-MuiGrid-root"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1l5bmg7-MuiPaper-root-MuiCard-root"
      >
        <div
          className="MuiCardMedia-root css-1r3z7eh-MuiCardMedia-root"
          role="img"
          style={
            {
              "backgroundImage": "url("https://cdn.chec.io/merchants/56103/assets/6uEp0drDg6pAUn5h|macbook.jpg")",
            }
          }
          title="Apple MacBook Air"
        />
        <div
          className="MuiCardContent-root css-2r3irp-MuiCardContent-root"
        >
          <div
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          >
            Apple MacBook Air
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-lw5d9e-MuiTypography-root"
          >
            $1,011.00
          </p>
        </div>
        <div
          className="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-m6jdh3-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                />
              </svg>
            </span>
            Add to cart
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root css-d4wxjj"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1oltwhr-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          <div
            className="MuiBox-root css-xi606m"
            data-testid="copyright-icon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CopyrightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          <div
            className="MuiBox-root css-xi606m"
          >
            Made by BTM INDUSTRIES
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
